jobs/integration/validation.py

```diff
@@ -2085,104 +2085,128 @@ async def test_multus(model, tools, addons_model):
+            "/snap/bin/kubectl --kubeconfig /root/.kube/config delete subnet attachnet --ignore-not-found;",
+        )
+        await multus_app.reset_config(["network-attachment-definitions"])
+        await containerd_app.reset_config(["no_proxy"])
+
+    async def apply_def(content):
+        remote_path = "/tmp/content.yaml"
+        with NamedTemporaryFile("w") as f:
+            yaml.dump(content, f)
+            await scp_to(
+                f.name, unit, remote_path, tools.controller_name, tools.connection
+            )
+        await run_until_success(
+            unit,
+            f"/snap/bin/kubectl --kubeconfig /root/.kube/config apply -f {remote_path}; rm {remote_path}",
         )
-        await multus_app.set_config({"network-attachment-definitions": ""})
 
     await cleanup()
 
-    # Create NetworkAttachmentDefinition for Flannel
+    await containerd_app.set_config(
+        {
+            "no_proxy": "localhost,127.0.0.1,::1,10.246.154.0/24,10.152.183.0/24,192.168.0.0/16"
+        }
+    )
+    await model.wait_for_idle(apps=["containerd"], timeout=5 * 60)
+
+    # Create NetworkAttachmentDefinition for KubeOVN
     net_attach_def = {
         "apiVersion": "k8s.cni.cncf.io/v1",
         "kind": "NetworkAttachmentDefinition",
-        "metadata": {"name": "flannel", "namespace": "default"},
+        "metadata": {"name": "attachnet", "namespace": "default"},
         "spec": {
             "config": json.dumps(
-                {
-                    "cniVersion": "0.3.1",
-                    "plugins": [
-                        {
-                            "type": "flannel",
-                            "delegate": {"hairpinMode": True, "isDefaultGateway": True},
-                        },
-                        {
-                            "type": "portmap",
-                            "capabilities": {"portMappings": True},
-                            "snat": True,
-                        },
-                    ],
-                }
+                dict(
+                    cniVersion="0.3.0",
+                    type="kube-ovn",
+                    server_socket="/run/openvswitch/kube-ovn-daemon.sock",
+                    provider="attachnet.default.ovn",
+                )
             )
         },
     }
     await multus_app.set_config(
         {"network-attachment-definitions": yaml.safe_dump(net_attach_def)}
     )
 
-    # Create pod with 2 extra flannel interfaces
+    # Create new subnet in OVN
+    attachnet_def = dict(
+        apiVersion="kubeovn.io/v1",
+        kind="Subnet",
+        metadata=dict(name="attachnet"),
+        spec=dict(
+            protocol="IPv4",
+            cidrBlock="10.166.0.0/16",
+            default=False,
+            excludeIps=["10.166.0.1"],
+            gateway="10.166.0.1",
+            gatewayType="distributed",
+            natOutgoing=True,
+            namespaces=["default"],
+        ),
+    )
```

Comment:
I'm a little puzzled that this subnet does not have the `provider` field set, which should be set to `attachnet.default` according to the Kube-OVN Multi-NIC docs [here](https://kubeovn.github.io/docs/v1.10.x/en/advance/multi-nic/#create-a-kube-ovn-subnet). But I see this subnet is copied from the Multus charm integration tests which haven't encountered any issues with the subnet, so... I guess it's not required? Seem weird, but, if it's working then I won't ask for any change here.